resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v2/Client.scala

```diff
@@ -0,0 +1,249 @@
+          .endMetadata()
+        .withNewSpec()
+          .addToContainers(driverContainer)
+          .endSpec()
+
+      val nonDriverPodKubernetesResources = mutable.Buffer[HasMetadata]()
+      val resolvedJars = mutable.Buffer[String]()
+      val resolvedFiles = mutable.Buffer[String]()
+      val driverPodWithMountedDeps = maybeStagingServerUri.map { stagingServerUri =>
+        val mountedDependencyManager = mountedDependencyManagerProvider.getMountedDependencyManager(
+          kubernetesAppId,
+          stagingServerUri,
+          allLabels,
+          namespace,
+          sparkJars,
+          sparkFiles)
+        val jarsResourceIdentifier = mountedDependencyManager.uploadJars()
+        val filesResourceIdentifier = mountedDependencyManager.uploadFiles()
+        val initContainerKubernetesSecret = mountedDependencyManager.buildInitContainerSecret(
+          jarsResourceIdentifier.resourceSecret, filesResourceIdentifier.resourceSecret)
+        val initContainerConfigMap = mountedDependencyManager.buildInitContainerConfigMap(
+          jarsResourceIdentifier.resourceId, filesResourceIdentifier.resourceId)
+        resolvedJars ++= mountedDependencyManager.resolveSparkJars()
+        resolvedFiles ++= mountedDependencyManager.resolveSparkFiles()
+        nonDriverPodKubernetesResources += initContainerKubernetesSecret
+        nonDriverPodKubernetesResources += initContainerConfigMap
+        mountedDependencyManager.configurePodToMountLocalDependencies(
+          driverContainer.getName, initContainerKubernetesSecret, initContainerConfigMap, basePod)
+      }.getOrElse {
+        sparkJars.map(Utils.resolveURI).foreach { jar =>
+          require(Option.apply(jar.getScheme).getOrElse("file") != "file",
+            "When submitting with local jars, a resource staging server must be provided to" +
+              s" deploy your jars into the driver pod. Cannot send jar with URI $jar.")
+        }
+        sparkFiles.map(Utils.resolveURI).foreach { file =>
+          require(Option.apply(file.getScheme).getOrElse("file") != "file",
+            "When submitting with local files, a resource staging server must be provided to" +
+              s" deploy your files into the driver pod. Cannot send file with URI $file")
+        }
+        resolvedJars ++= sparkJars
+        resolvedFiles ++= sparkFiles
+        basePod
+      }
+      val resolvedSparkConf = sparkConf.clone()
+      if (resolvedJars.nonEmpty) {
+        resolvedSparkConf.set("spark.jars", resolvedJars.mkString(","))
+      }
+      if (resolvedFiles.nonEmpty) {
+        resolvedSparkConf.set("spark.files", resolvedFiles.mkString(","))
+      }
+      resolvedSparkConf.set(KUBERNETES_DRIVER_POD_NAME, kubernetesAppId)
+      resolvedSparkConf.set("spark.app.id", kubernetesAppId)
+      // We don't need this anymore since we just set the JVM options on the environment
+      resolvedSparkConf.remove(org.apache.spark.internal.config.DRIVER_JAVA_OPTIONS)
```

Comment:
The problem is that since we pass these in verbatim to the driver JVM, we would have to escape these in order for the Java invocation to treat this properly as a single string.